File: dart/test/all_registered_test.dart
/// All Registered Domains test for SNS SDK
/// Mirrors js/tests/all-registered.test.ts exactly
library;

import 'package:dotenv/dotenv.dart';
import 'package:sns_sdk/sns_sdk.dart';
import 'package:test/test.dart';

void main() {
  group('All registered domains', () {
    test('All registered', () async {
      // Load environment variables from .env file
      final env = DotEnv()..load(['.env']);
      final rpcUrl = env['RPC_URL'];

      if (rpcUrl == null || rpcUrl.isEmpty) {
        markTestSkipped('RPC_URL environment variable not set');
        return;
      }

      // Create connection with extended timeout for this intensive operation
      final connection = EnhancedSolanaRpcClient(
        rpcUrl,
        timeout: const Duration(minutes: 4), // Match the test timeout
      );

      // Get all registered domains
      final registered = await getAllRegisteredDomains(connection);
      expect(registered.length, greaterThan(130000));
    }, timeout: const Timeout(Duration(minutes: 4)));
  });
}

File: js/tests/sns/all-registered.test.ts
require("dotenv").config();
import { Connection } from "@solana/web3.js";
import { test, expect, jest } from "@jest/globals";
import { getAllSnsDomains } from "../../src/utils/getAllSnsDomains";

jest.setTimeout(4 * 60_000);

const connection = new Connection(process.env.RPC_URL!);

test("All registered", async () => {
  const registered = await getAllSnsDomains(connection);
  expect(registered.length).toBeGreaterThan(130_000);
});
